src/ceph_osd.cc

```diff
@@ -356,6 +361,21 @@ int main(int argc, const char **argv)
 	derr << "created new key in keyring " << keyring_path << dendl;
     }
   }
+
+  if (!osdspec_affinity.empty()) {
```

Comment:
it looks like the intent is to only use this option with --mkfs. If so, I'd suggest passing it through OSD::mkfs() to OSD::write_meta() so that all the metadata is recorded in one place. That way we don't record osdspec_affinity if there's an error during mkfs as well.

Reply:
agreed & done